### PR TITLE
Bugfixes: GET User Permissions

### PR DESCRIPTION
**Pull request by EMcQ-BES**
### Changes:

- Fixup column handling to use resource/record conversion
- Add missing endpoint for user/id/userEntityPermission
- Fully justify table names in permission filters

### CODE REVIEW

File: packages/meditrak-server/src/routes/GETSurveyScreenComponents/assertSurveyScreenComponentPermissions.js [hunk 1]
```diff
@@ -34,17 +34,23 @@ export const createSurveyScreenComponentDBFilter = async (
...
-  const dbConditions = { ...criteria };
+  let dbConditions = { ...criteria };
   const dbOptions = { ...options };
 
   if (surveyId) {
-    dbConditions.survey_id = surveyId;
+    dbConditions = {
+      'survey_screen.survey_id': surveyId,
+      ...criteria,
+    };
```
Comment (kael89) [inline packages/meditrak-server/src/routes/GETSurveyScreenComponents/assertSurveyScreenComponentPermissions.js:44]: This way you can still use a `const` in line 37 above, and it's a bit simpler

```suggestion
    dbConditions['survey_screen.survey_id'] = surveyId;
```

Unless if you expect that a `survey_screen.survey_id` may exist in `criteria`, and if it exists you want it to take precedence over the `surveyId` value

File: packages/meditrak-server/src/routes/GETSurveyScreenComponents/assertSurveyScreenComponentPermissions.js [hunk 2]
```diff
@@ -34,17 +34,23 @@ export const createSurveyScreenComponentDBFilter = async (
...
   } else if (!hasBESAdminAccess(accessPolicy)) {
     // If we have BES admin, don't bother filtering by survey
     const surveyConditions = await createSurveyDBFilter(accessPolicy, models);
     const permittedSurveys = await models.survey.find(surveyConditions);
     const permittedSurveyIds = permittedSurveys.map(s => s.id);
-    dbConditions.survey_id = permittedSurveyIds;
+    dbConditions = {
+      'survey_screen.survey_id': permittedSurveyIds,
+      ...criteria,
```
Comment (kael89) [inline packages/meditrak-server/src/routes/GETSurveyScreenComponents/assertSurveyScreenComponentPermissions.js:52]: Same as above

**Review by kael89 — CHANGES_REQUESTED** [review]
Looks great overall @EMcQ-BES ! Just one small suggestion

**Comment by kael89** [comment]
@EMcQ-BES  There's also a failing unit test

**Comment by EMcQ-BES** [comment]
> @EMcQ-BES There's also a failing unit test

Unit test was failing because of filters overwriting each other, I've added a function to merge filters together

File: packages/meditrak-server/src/routes/utilities/mergeFilter.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+
+export function mergeFilter(baseFilter, filterToMerge) {
```
Comment (kael89) [inline packages/meditrak-server/src/routes/utilities/mergeFilter.js:6]: Nice util!

**Review by kael89 — APPROVED** [review]
Looks good!

**Pull request merged**